feat: feature/promote php actions

sdh100shaun opened the pull request:
## Description

Add in actions for PHP

Related issue: N/A

## Before submitting (or marking as "ready for review")

- [x] Does the pull request title follow the [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/) specification?
- [x] Have you performed a self-review of the code
- [ ] Have you have added tests that prove the fix or feature is effective and working
- [x] Did you make sure to update any documentation relating to this change?

jerotire commented on .github/workflows/php-security.yml:
```diff
@@ -0,0 +1,47 @@
+
+    strategy:
+      matrix:
+        stability: [prefer-lowest, prefer-stable]
```
Is it also worth adding another matrix param for PHP versions? 8.0, 8.1 and 8.2? I know we do so for tests, but wondering if we need for security and static too?

AnnaDodson commented:
Because the `on` keyword is used, the actions are getting ran on the PR and then failing - should these be in the workflow templates directory instead? Or split so they are just reusable workflows and then add a template to show an example of it working - see https://docs.github.com/en/actions/using-workflows/reusing-workflows

sdh100shaun commented:
updated and added matrix for them all - see https://github.com/dvsa/dvsa-php-action-tester for these working in a php project configured as expected - not there are some linting failures and test failures but that is the project - once happy will need to change the branch name on that project - but please see what you think @AnnaDodson  @jerotire  then I will do that and add a readme

jerotire commented on .github/workflows/php-static.yml:
```diff
@@ -0,0 +1,96 @@
+        with:
+          php-version: ${{ matrix.php }}
+          coverage: none
+          extensions: mbstring, intl
```
Do we need to accept this as an definable input with a sane default? I know we use other extensions in VOL for instance.

sdh100shaun replied:
As discussed the other day I think for test it is useful

sdh100shaun commented on .github/workflows/php-security.yml:
```diff
@@ -0,0 +1,48 @@
+   runs-on: ubuntu-latest
+   strategy:
+      matrix:
+        php: ${{ fromJSON(inputs.php-versions) }}
```
fair point - yes we don't care right

sdh100shaun commented on .github/workflows/php-security.yml:
```diff
@@ -0,0 +1,48 @@
+        args: --sarif-file-output=snyk.sarif
+
+    - name: Upload result to GitHub Code Scanning
+      if: ${{ github.action_repository.private == false && (success() || failure() )}}
```
yeah I think github.repository.private is what I was looking for ,  so this was to ensure we can run with or without github avdanced security

JoshuaLicense replied:
Still an issue ^

JoshuaLicense replied:
I don't believe this will work when triggered by a `schedule` event.

sdh100shaun commented on .github/workflows/php-static.yml:
```diff
@@ -0,0 +1,96 @@
+  phpstan:
+    strategy:
+      matrix:
+        php: ${{ fromJSON(inputs.php-versions) }}
```
so what do we want to do here ? @jerotire  thoughts having proposed we have the matrix ? Did you envisage us having some conditional rules ?  I am not sure I really questioned it

jerotire replied:
Yeah, @JoshuaLicense makes sense. No need for different PHP versions for static analysis. I do not envisage us having conditional rules in the future, perhaps a little now since some projects are using different versions of PHP which some later features being unsupported in our current versions. But, that should not be an issue later right? 😄

sdh100shaun replied:
build for the now 👍

sdh100shaun commented on .github/workflows/php-static.yml:
```diff
@@ -0,0 +1,96 @@
+    strategy:
+      matrix:
+        php: ${{ fromJSON(inputs.php-versions) }}
+        stability: [prefer-lowest, prefer-stable]
```
er not here .. good catch

sdh100shaun commented on .github/workflows/php-static.yml:
```diff
@@ -0,0 +1,96 @@
+        run: composer update --no-interaction --no-progress
+
+      - name: Install PHPStan
+        run: composer bin phpstan update --no-interaction --no-progress
```
so this is interesting ... I adopted this the auth repo and I had a few questions in my head .. I was not aware of this ..The setup-php action now has these tools built in as parameters,

sdh100shaun commented on .github/workflows/php-tests.yml:
```diff
@@ -0,0 +1,36 @@
+        with:
+          php-version: ${{ matrix.php }}
+          tools: composer:v2
+          coverage: none
```
makes sense here I think

sdh100shaun commented on .github/workflows/php-security.yml:
```diff
@@ -0,0 +1,48 @@
+      with:
+        php-version: ${{ matrix.php }}
+        coverage: none
+        extensions: mbstring
```
have left this out till we see a use case

sdh100shaun commented on .github/workflows/php-static.yml:
```diff
@@ -0,0 +1,62 @@
+ workflow_call:
+
+jobs:
+  phpstan:
```
good catch

sdh100shaun commented:
@AnnaDodson  @JoshuaLicense  @jerotire  this is ready for review and merging
